File: src/lib/debug.ts
const isDebugEnabled = () =>
  Boolean((import.meta as ImportMeta & { env?: { DEV?: boolean } }).env?.DEV) ||
  localStorage.getItem("qpaw:debug") === "1";

export function debugLog(scope: string, details?: Record<string, unknown>) {
  if (!isDebugEnabled()) return;
  console.debug(`[QPaw debug][${scope}]`, details ?? {});
}

export function debugError(scope: string, error: unknown, details?: Record<string, unknown>) {
  if (!isDebugEnabled()) return;
  console.error(`[QPaw error][${scope}]`, { ...details, error: formatError(error) });
}

export function formatError(error: unknown) {
  return error instanceof Error ? error.message : String(error);
}

File: src-tauri/src/debug.rs
pub fn log(scope: &str, message: impl AsRef<str>) {
    #[cfg(debug_assertions)]
    eprintln!("[QPaw debug][{scope}] {}", message.as_ref());
}

pub fn err(scope: &str, message: impl AsRef<str>) {
    #[cfg(debug_assertions)]
    eprintln!("[QPaw error][{scope}] {}", message.as_ref());
}
